### PR TITLE
Network Refactor

### PR DESCRIPTION
**Pull request by tli2**
We screwed up some dependencies in the network layer. I had to shuffle things around to make sure we are not hardcoding postgres dependencies in the general network layer.

This led to the discovery of #413. Some more work remains to be done to rip the dispatcher and handler creation out of the terrier server, so that in the future we can easily have multiple dispatchers and handlers with heterogenous protocols. Not done in this PR because it probably conflicts with @GustavoAngulo

### CODE REVIEW

File: src/include/network/protocol_interpreter.h [hunk 1]
```diff
@@ -14,6 +15,10 @@ class ConnectionHandle;
  */
 class ProtocolInterpreter {
  public:
+  struct Provider {
```
Comment (apavlo) [inline src/include/network/protocol_interpreter.h]: Add comment to explain what this is used for.

File: test/include/util/manual_packet_util.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/*
+ * This header includes some handy helper functions when testing with manual constructed packets.
```
Comment (apavlo) [inline test/include/util/manual_packet_util.h]: Move this to the be the description of the util class.

File: test/include/util/manual_packet_util.h [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "loggers/test_logger.h"
+#include "network/connection_handle_factory.h"
+#include "util/manual_packet_util.h"
+
+namespace terrier::network {
+struct ManualPacketUtil {
```
Comment (apavlo) [inline test/include/util/manual_packet_util.h]: Why is this is a struct and not a class like the other util classes?

File: src/include/network/network_defs.h [hunk 4]
```diff
@@ -23,7 +23,7 @@
 #include "traffic_cop/result_set.h"
 #include "type/type_id.h"
 
-namespace terrier::traffic_cop {
+namespace terrier::tcop {
```
[the remark this answers is not in the document]
Comment (tli2) [reply src/include/network/network_defs.h]: Namespaces are not supposed to have underscores, and I thought `trafficcop` looked weird.

Can change if you don't like that

**Pull request merged**